doc/dash-acl-tags/dash-acl-tags.md

```diff
@@ -0,0 +1,277 @@
+
+### Runtime
+
+When an update for the DASH_PREFIX_TAG_TABLE or DASH_ACL_RULE_TABLE table is received DashAclOrch will understand the operation type. The supported operation types are create, update, and remove.
+
+### Tags handling
+
+
+**Add new ACL tag**
+
+If the operation is to create a new ACL tag in DASH_PREFIX_TAG_TABLE DashAclOrch will do the following
+- Validate the tag configuration
+- Store the tag configuration
+
+**Remove ACL tag**
+
+If the operation is to remove the ACL tag from DASH_PREFIX_TAG_TABLE DashAclOrch will do the following
+- If the tag is attached to the rules
+    - Return status "retry" to postpone tag removal
+- Remove the tag
+
+**Update ACL tag**
+
+If the operation is to update the list of prefixes in the ACL tag in DASH_PREFIX_TAG_TABLE DashAclOrch will do the following
+  - Based on the list of ACL rules that uses the tag get list of affected ACL groups
+  - For each affected ACL group
+  - If group is bound to the ENI
+    - Create new ACL group
+    - For each ACL rule from the original ACL group
+      - Copy original ACL rule attributes
+      - Update attributes if required
+      - Create new ACL rule and attach it to the new ACL group
+    - Bind new ACL group to the ENI
+    - Remove all old ACL rules
+    - Remove old ACL group
+  - Else (if ACL group is not bound)
+    - For each ACL rule that uses the tag
+      - Update ACL rules
+
+### ACL rules handling
+
+
+**ACL rule creation flow**
+
+If the operation is to create a new ACL rule in DASH_ACL_RULE_TABLE DashAclOrch will do the following
+- If tags are specified in the configuration
+    - For each configured tag
+        - Check if the tags exists in `DASH_PREFIX_TAG_TABLE`. If the tag doesn't exist return status "retry" to postpone rule creation.
+    - If all tags are available
+      - Expand all tags to a list of prefixes
+      - Create ACL rule
+
+**Remove ACL rule**
+
+If the operation is to remove the existing ACL rule from DASH_ACL_RULE_TABLE DashAclOrch will do the following
+- Remove rule
+
+**Update ACL rule**
+
+If the operation is to update the existing ACL rule in DASH_ACL_RULE_TABLE DashAclOrch will do the following
+- if the ACL group to which the ACL rule belongs is bound to the ENI
+  - Create a new ACL group
+  - For each ACL rule from the original ACL group
+    - Copy original ACL rule attributes
+    - Update attributes if required
+    - Create a new ACL rule and attach it to the new ACL group
+- Else
+  - For each changed attribute
+    - Update attribute configuration
+
+# SAI API
+
+Stage 1 implementation does not require SAI API.
+
+# Configuration and management
+
+## CLI/YANG model Enhancements
+
+No CLI/YANG enhancements required.
+
+## Config DB Enhancements
+
+No Config DB Enhancements required
+
+# Warmboot and Fastboot Design Impact
+DPU SONiC doesn't support warm/fast boot
+
+# Restrictions/Limitations
```

Comment:
Should we mention how we handle tags with overlapping prefixes? I prefer we capture the handling in some section and update the limitations if any here